main.tf

```diff
@@ -1,92 +1,75 @@
+    role_id     = var.role_id == null ? "SpotRole${replace(spotinst_account.spot_acct.id, "-", "")}" : var.role_id
+    title       = var.role_title == null ? "SpotRole${replace(spotinst_account.spot_acct.id, "-", "")}" : var.role_title
+    description = var.role_description
+    project     = var.project
+    permissions = var.role_permissions
 }
 
 resource "google_service_account" "spotserviceaccount" {
-	provisioner "local-exec" {
-	    # Without this set-cloud-credentials fails 
-	    command = "sleep 10"
-	}
-	account_id   = "spot-${random_id.role.hex}"
-	display_name = "spot-${random_id.role.hex}"
-	description = "Service Account for Spot.io"
-	project = var.project
+    depends_on = [spotinst_account.spot_acct]
+    provisioner "local-exec" {
+        # Without this set-cloud-credentials fails
+        command = "sleep 10"
+    }
+    account_id   = var.service_account_id == null ? "spot-${var.spot_organization_id}-${spotinst_account.spot_acct.id}" : var.service_account_id
+    display_name = var.service_account_display_name == null ? "spot-${var.spot_organization_id}-${spotinst_account.spot_acct.id}" : var.service_account_display_name
+
+    description  = var.service_account_description
+    project      = var.project
 }
 
 resource "google_service_account_key" "key" {
-  service_account_id = google_service_account.spotserviceaccount.name
+    service_account_id = google_service_account.spotserviceaccount.name
 }
 
 resource "google_project_iam_binding" "spot-account-iam" {
     project = var.project
-    role = google_project_iam_custom_role.SpotRole.name
+    role    = google_project_iam_custom_role.SpotRole.name
     members = [
-        "serviceAccount:spot-${random_id.role.hex}@${var.project}.iam.gserviceaccount.com",
+        google_service_account.spotserviceaccount.member
```

Comment:
Is this member's change correct?

Reply:
Yes, followed the same ways as used in gcp-connect module.